### DISCUSSION TITLE
Create a network from oemof.dhnx network data or geodata with "from-to" information of nodes and pipes

### DISCUSSION THREAD

**Discussion by gitfeldhase** [e2nIEE/pandapipes#438] [General]
Hello everyone,

I am a completely new to pandapipes and I am wondering if it is possible to read in a network from a folder structure like the following:
[network_folder.zip](https://github.com/e2nIEE/pandapipes/files/10235764/network_folder.zip)

In this network I have the information on the location of the consumers and producers, the lengths of the pipes connecting all together with id's and a "from-to" structure similiar to a graph.

I checkes the tutorials but couldn't find one where external data was used to create a piping network.

Best regards and thanks for the support!

**Comment by dlohmeier** [comment]
Dear @gitfeldhase ,

welcome to the pandapipes repo. The conversion of network data from different input formats has a long history in the development of pandapipes and especially [pandapower](https://github.com/e2nIEE/pandapower/tree/develop/pandapower/converter). The sad news is that there is no direct solution like one command and tada, here's your pandapipes network. The good news is that with the help of pandas and the nice [bulk-create functions](https://github.com/e2nIEE/pandapipes/blob/13786725165adb5a45f2dc694c59e224616ba7e5/pandapipes/create.py#L821) provided in pandapipes, you should have an easy time to write your own converter. We are currently working on providing a STANET converter (developments, see [here](https://github.com/e2nIEE/pandapipes/tree/stanet_conversion/pandapipes/converter/stanet)), from which you might get an idea of what to do. So basically, the procedure would be:
1. Load your CSV-files as pandas tables.
2. Use a bulk create-function (or loop over the normal create functions) for the respective component you want to create and hand over the correct columns as entries. Please note that you will always have to create a correct branch-node-model, i.e. we explicitly have to define all nodes (junctions). In your case that means that you have to create junctions for all producers, consumers and forks first, and then add the node elements sink (for consumers) and source (for producers) to add injection / extraction data to those nodes. If you want to model time series, you need to add controllers in order to influence the value of mdot_kg_per_s for each time step.
3. Check if any important data is missing (one important aspect is the constant pressure / temperature node, which in pandapipes is modeled by an external grid), and add it to your script.
4. Check and test the resulting network, some fine-tuning might include pipeflow settings or specific settings for the nominal junction pressures.

If you encounter any problems, just let us know. Btw. is that some standard formulation like GasLib or SciGridGas? As such a converter would be useful for many others, we would like to encourage you to contribute to pandapipes in order to add more converters to our library.

Kind regards

**Comment by gitfeldhase** [comment]
Hello,

thanks a lot for the fast reply and the detailed descriptions!
I will have a look at the _create functions_.

The dataformat is the output of the oemof.dhnx library which is quite nice to creat new grids from geodata from osm.
Though, I think the simulation module of pandapipes offers more possibilities than the simulatiom module of DHNx.
Coupling both and using the strengths of both repo's would be nice :) 

Thanks for the support!

**Comment by dlohmeier** [comment]
Ah, it's oemof, all right. Seemed familiar to me. In that case, a converter would be really nice to have. Thanks for the hint.
